botocove/cove_host_account.py

```diff
@@ -110,11 +111,97 @@ def _resolve_target_accounts(self, target_ids: Optional[List[str]]) -> Set[str]:
             # No target_ids passed
             target_accounts = self._gather_org_assume_targets()
         else:
-            # Specific list of IDs passed
-            target_accounts = set(target_ids)
+            target_accounts, target_ous = self._validate_target_accounts(target_ids)
+            target_accounts_from_ous = self._get_all_accounts_by_organization_units(
+                target_ous
+            )
+            target_accounts.extend(target_accounts_from_ous)
+
+            target_accounts = set(target_accounts)
 
         return target_accounts - validated_ignore_ids
 
+    def _validate_target_accounts(
+        self, target_ids: Optional[List[str]]
+    ) -> Tuple[List[str], List[str]]:
+
+        target_accounts = list()
+        target_ous = []
+
+        for target_id in target_ids:
+            if not isinstance(target_id, str):
+                raise TypeError("All target_id list entries must be strings")
+            if not re.match(r"^\d{12}$", target_id):
+                if not re.match(r"^ou-[0-9a-z]{4,32}-[a-z0-9]{8,32}$", target_id):
+                    raise TypeError(
+                        f"target_ids entry is neither an aws account nor an ou: {target_id}"
+                    )
+                else:
+                    target_ous.append(target_id)
+            else:
+                target_accounts.append(target_id)
```

Comment:
I'd prefer we avoid nested if/else/else logic: how about this?
nit: changed TypeError to ValueError

```suggestion
        for target_id in target_ids:
            if not isinstance(target_id, str):
                raise TypeError("All target_id list entries must be strings")
            if re.match(r"^\d{12}$", target_id):
                target_accounts.append(target_id)
                continue
            if re.match(r"^ou-[0-9a-z]{4,32}-[a-z0-9]{8,32}$", target_id):
                target_ous.append(target_id)
                continue
            raise ValueError(
                f"target_ids entry is neither an aws account nor an ou: {target_id}"
            )
```